src/neuroconv/datainterfaces/behavior/video/externalvideointerface.py

```diff
@@ -315,6 +333,15 @@ def add_to_nwbfile(
             videos_metadata = deepcopy(self.get_metadata()["Behavior"]["ExternalVideos"])
         image_series_kwargs = videos_metadata[self.video_name]
         image_series_kwargs["name"] = self.video_name
+        device_kwargs = image_series_kwargs.pop("device", None)
```

Comment:
OK, one last question. If the user passes metadata but doesn't include the nested `device` dictionary, then no device will be created, right? Is that the intended behavior?

There are also a couple of edge cases, for example, if the user provides the `device` dictionary but not the `name`. I think this will be caught by the doc_val or the `validate_metadata` at write time. What do you think?

Reply:
> There are also a couple of edge cases, for example, if the user provides the device dictionary but not the name. I think this will be caught by the doc_val or the validate_metadata at write time. What do you think?

Yep, this gets caught. I just added a test to ensure this behavior.

Reply:
Good. It makes sense to have a way of doing that.